jenkins/scripts/VersionSelectorScript.groovy

```diff
@@ -0,0 +1,94 @@
+def canBuild(nodeMajorVersion, builderLabel, buildType) {
+
+  // Linux
+  if (builderLabel.indexOf('centos5') == 0 && nodeMajorVersion >= 8)
+    return false
+  if (buildType == 'release' && builderLabel.indexOf('centos6') == 0 && nodeMajorVersion < 8)
+    return false
+  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc48/ && nodeMajorVersion >= 10)
+    return false
+  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc6/ && nodeMajorVersion < 10)
+    return false
+  if (builderLabel.indexOf('ubuntu1804') == 0 && nodeMajorVersion < 10) // probably temporary
+    return false
+  if (builderLabel.indexOf('ubuntu1204') == 0 && nodeMajorVersion >= 10)
+    return false
+
+  // Windows
+  if (builderLabel.indexOf('vs2013-') == 0 && nodeMajorVersion >= 6)
+    return false
+  if (builderLabel.indexOf('vs2015-') == 0 && (nodeMajorVersion < 6 || nodeMajorVersion >= 10))
+    return false
+  if (builderLabel.indexOf('vs2017-') == 0 && nodeMajorVersion < 10)
+    return false
+
+  // SmartOS
+  if (builderLabel.indexOf('smartos13') == 0) // Node.js 0.x
+    return false
+  if (builderLabel.indexOf('smartos14') == 0 && nodeMajorVersion >= 8)
+    return false
+  if (builderLabel.indexOf('smartos15') == 0 && nodeMajorVersion < 8)
+    return false
+
+  if (builderLabel.equals('debian7-docker-armv7') && nodeMajorVersion >= 10)
+    return false
+  if (buildType == 'release' && builderLabel.equals('debian8-docker-armv7') && nodeMajorVersion < 10)
+    return false
+  if (builderLabel.equals('debian9-docker-armv7') && nodeMajorVersion < 10)
+    return false
+
+  // PPC BE
+  if (builderLabel.indexOf('ppcbe-ubuntu') == 0 && nodeMajorVersion >= 8)
+    return false
+
+  // s390x
+  if (builderLabel.indexOf('s390x') > -1 && nodeMajorVersion < 6)
+    return false
+
+  // AIX61
+  if (builderLabel.indexOf('aix61') > -1 && nodeMajorVersion < 6)
+    return false
+
+  // sharedlibs containered
+  if (builderLabel.indexOf('sharedlibs_openssl111') > -1 && nodeMajorVersion < 9)
+    return false
+  if (builderLabel.indexOf('sharedlibs_openssl110') > -1 && nodeMajorVersion < 9)
+    return false
+  if (builderLabel.indexOf('sharedlibs_openssl102') > -1 && nodeMajorVersion > 9)
+    return false
+  if (builderLabel.indexOf('sharedlibs_fips20') > -1 && nodeMajorVersion > 9)
+    return false
+  if (builderLabel.indexOf('sharedlibs_withoutintl') > -1 && nodeMajorVersion < 9)
+    return false
+
+  return true
+}
+
+// setup for execution of the above rules
+
+int nodeMajorVersion = -1
```

Comment:
So this will just skip everything if something goes wrong right? If that happens then that means the build will be green right?

It'd be nice to fail the build if we get an unexpected Node version, maybe just ensure it's in a certain range, and then throw if not.

That would also remind us to update the supported platforms list before a new Node version comes out, and give it some urgency.